app/src/main/java/com/vero/tasky/auth/data/repository/AuthRepositoryImpl.kt

```diff
@@ -0,0 +1,58 @@
+package com.vero.tasky.auth.data.repository
+
+import com.vero.tasky.auth.data.dto.toUser
+import com.vero.tasky.auth.data.network.AuthApi
+import com.vero.tasky.auth.data.request.LoginRequest
+import com.vero.tasky.auth.data.request.RegisterRequest
+import com.vero.tasky.auth.domain.repository.AuthRepository
+import com.vero.tasky.core.domain.model.User
+
+class AuthRepositoryImpl(
+    private val api : AuthApi
+) : AuthRepository {
+
+    override suspend fun register(
+        email: String,
+        password: String,
+        fullName: String
+    ): Result<Unit> {
+        return try {
+            api.register(
+                RegisterRequest(
+                    fullName = fullName,
+                    password = password,
+                    email = email
+                )
+            )
+            Result.success(Unit)
+        } catch (e: Exception) {
+            e.printStackTrace()
+            Result.failure(e)
```

Comment:
There's one major problem when using a simple try catch

You are also catching the exception called "CancellationException"
So if a coroutine get's canceled (user leaves the screen for example), then the coroutine won't be successfully cancelled

To fix this issue, you could:

```
return try {
        api.register(
            RegisterRequest(
                fullName = fullName,
                password = password,
                email = email
            )
        )
        Result.success(Unit)
    } catch (e: CancellationException) {
        throw e
    } catch (e: Exception) {
        e.printStackTrace()
        Result.failure(e)
}
```

In fact, to avoid having to create this try catch all the time, you can create an extension function ;)